admin/class-primary-term-admin.php

```diff
@@ -42,57 +39,18 @@ protected function get_current_id() {
 
 	/**
 	 * Adds hidden fields for primary taxonomies.
+	 * Post type parameter is not used, but is provided by the filter.
 	 *
-	 * @param string $content The metabox content.
+	 * @param array<array<string>> $field_defs The fields defs for general group.
 	 *
-	 * @return string The HTML content.
+	 * @return array<array<string>> The new general group fields defs.
 	 */
-	public function add_input_fields( $content ) {
+	public function add_input_fields( $field_defs ) {
 		$taxonomies = $this->get_primary_term_taxonomies();
-
 		foreach ( $taxonomies as $taxonomy ) {
-			$content .= $this->primary_term_field( $taxonomy->name );
-			$content .= wp_nonce_field( 'save-primary-term', WPSEO_Meta::$form_prefix . 'primary_' . $taxonomy->name . '_nonce', false, false );
+			$field_defs[ 'primary_' . $taxonomy->name ] = WPSEO_Meta::$meta_fields['primary_terms'][ 'primary_' . $taxonomy->name ];
```

Comment:
What if the taxonomy didn't exist yet in the initializer?

Reply:
Moved the registration to the register taxonomy hook